Libs/MRML/Logic/vtkMRMLSliceLogic.cxx

```diff
@@ -1958,82 +1949,55 @@ void vtkMRMLSliceLogic::ResizeSliceNode(double newWidth, double newHeight)
 //----------------------------------------------------------------------------
 double *vtkMRMLSliceLogic::GetLowestVolumeSliceSpacing()
 {
-  // TBD: Doesn't return the lowest slice spacing, just the first valid spacing
-  vtkMRMLVolumeNode *volumeNode;
-  for ( int layer=0; layer < 3; layer++ )
+  // Note that "lowest" refers to the lowest (first) volume layer, not the one with the lowest slice spacing
+  vtkMRMLVolumeNode* volumeNode = vtkMRMLSliceLogic::GetFirstVolumeNode();
+  if (volumeNode)
   {
-    volumeNode = this->GetLayerVolumeNode (layer);
-    if (volumeNode)
-    {
-      return this->GetVolumeSliceSpacing( volumeNode );
-    }
+    return this->GetVolumeSliceSpacing(volumeNode);
   }
   return (this->SliceSpacing);
 }
 
 //----------------------------------------------------------------------------
 void vtkMRMLSliceLogic::GetLowestVolumeSliceBounds(double sliceBounds[6], bool useVoxelCenter/*=false*/)
 {
-  vtkMRMLVolumeNode *volumeNode;
-  for ( int layer=0; layer < 3; layer++ )
+  vtkMRMLVolumeNode* volumeNode = vtkMRMLSliceLogic::GetFirstVolumeNode();
+  if (volumeNode)
   {
-    volumeNode = this->GetLayerVolumeNode (layer);
-    if (volumeNode)
-    {
-      return this->GetVolumeSliceBounds(volumeNode, sliceBounds, useVoxelCenter);
-    }
+    return this->GetVolumeSliceBounds(volumeNode, sliceBounds, useVoxelCenter);
   }
   // return the default values
   return this->GetVolumeSliceBounds(nullptr, sliceBounds, useVoxelCenter);
 }
 
-#define LARGE_BOUNDS_NUM 1.0e10
-#define SMALL_BOUNDS_NUM -1.0e10
 //----------------------------------------------------------------------------
 void vtkMRMLSliceLogic::GetSliceBounds(double sliceBounds[6])
 {
-  int i;
-  for (i=0; i<3; i++)
-  {
-    sliceBounds[2*i]   = LARGE_BOUNDS_NUM;
-    sliceBounds[2*i+1] = SMALL_BOUNDS_NUM;
-  }
-
-  vtkMRMLVolumeNode *volumeNode;
-  for ( int layer=0; layer < 3; layer++ )
+  vtkBoundingBox sliceBoundingBox;
+  for (int layer=0; layer < vtkMRMLSliceLogic::Layer_Last; layer++)
```

Comment:
```suggestion
  for (int layer = 0; layer < Self::Layer_Last; layer++)
```

Reply:
I would keep the VTK style of using the actual class name